baseapp-cloudflare-stream-field/baseapp_cloudflare_stream_field/tasks.py

```diff
@@ -14,9 +15,36 @@ def refresh_from_cloudflare(content_type_pk, object_pk, attname, retries=1):
     cloudflare_video = getattr(obj, attname)
     if cloudflare_video["status"]["state"] != "ready":
         new_value = stream_client.get_video_data(cloudflare_video["uid"])
-        if new_value["status"]["state"] == "ready":
-            setattr(obj, attname, new_value)
-            obj.save(update_fields=[attname])
+        new_video = None
+
+        if (
+            new_value["status"]["state"] == "ready"
+            and new_value["status"]["errorReasonCode"] != "ERR_DURATION_EXCEED_CONSTRAINT"
+        ):
+
+            if hasattr(settings, "CLOUDFLARE_VIDEO_AUTOMATIC_TRIM") and hasattr(
+                settings, "CLOUDFLARE_VIDEO_MAX_DURATION_SECONDS"
+            ):
+                if new_value["clippedFrom"] is None:
+                    old_video_uid = new_value["uid"]
+                    new_video = stream_client.clip_video(
+                        {
+                            "clippedFromVideoUID": old_video_uid,
+                            "startTimeSeconds": 0,
+                            "endTimeSeconds": settings.CLOUDFLARE_VIDEO_MAX_DURATION_SECONDS,
+                        }
+                    )
+                if new_video:
+                    setattr(obj, attname, new_video)
+                    delete_original_trimmed_video.apply_async(
+                        kwargs={
+                            "old_video_uid": old_video_uid,
+                            "new_video_uid": new_video["uid"],
+                        },
+                    )
+                else:
+                    setattr(obj, attname, new_value)
```

Comment:
I think you need to change this if block here, seems like `setattr(obj, attname, new_value)` will never run if you don't have this settings, which would break old projects, I think what you could to do instead of `hasattr` is `getattr` for the trim setting:

```python
getattr(settings, "CLOUDFLARE_VIDEO_AUTOMATIC_TRIM", False) and hasattr(obj, "CLOUDFLARE_VIDEO_MAX_DURATION_SECONDS")
```

And have a `setattr(obj, attname, new_value)` executing as well if those settings are not set